homeassistant/components/generic/config_flow.py

```diff
@@ -0,0 +1,162 @@
+                )
+                video_stream = container.streams.video[0]
+                if video_stream is not None:
+                    return True
+            except av.error.HTTPUnauthorizedError:  # pylint: disable=c-extension-no-member
+                self._errors["base"] = "stream_unauthorised"
+            except (KeyError, IndexError):
+                self._errors["base"] = "stream_novideo"
+            except OSError as err:
+                if "No route to host" in str(err):
+                    self._errors["base"] = "stream_no_route_to_host"
+            return False
+        return True
+
+    @staticmethod
+    def build_schema(user_input):
+        """Create schema for camera config setup."""
+        spec = {
+            vol.Optional(CONF_NAME, default=user_input[CONF_NAME]): str,
+            vol.Required(
+                CONF_STILL_IMAGE_URL,
+                default=user_input.get(CONF_STILL_IMAGE_URL),
+            ): str,
+            vol.Optional(
+                CONF_STREAM_SOURCE,
+                default=user_input.get(CONF_STREAM_SOURCE, ""),
+            ): str,
+            vol.Optional(
+                CONF_RTSP_TRANSPORT,
+                default=user_input.get(CONF_RTSP_TRANSPORT, False),
+            ): bool,
+            vol.Optional(
+                CONF_AUTHENTICATION, default=user_input[CONF_AUTHENTICATION]
+            ): vol.In([HTTP_BASIC_AUTHENTICATION, HTTP_DIGEST_AUTHENTICATION]),
+            vol.Optional(
+                CONF_USERNAME,
+                default=user_input.get(CONF_USERNAME, ""),
+            ): str,
+            vol.Optional(CONF_PASSWORD, default=user_input.get(CONF_PASSWORD, "")): str,
+            vol.Optional(
+                CONF_LIMIT_REFETCH_TO_URL_CHANGE,
+                default=user_input[CONF_LIMIT_REFETCH_TO_URL_CHANGE],
+            ): bool,
+            vol.Optional(CONF_CONTENT_TYPE, default=user_input[CONF_CONTENT_TYPE]): str,
+            vol.Optional(CONF_FRAMERATE, default=user_input[CONF_FRAMERATE]): int,
+            vol.Optional(CONF_VERIFY_SSL, default=user_input[CONF_VERIFY_SSL]): bool,
+        }
+        return vol.Schema(spec)
+
+    async def async_step_user(
+        self, user_input: Dict[str, Any] = None
+    ) -> Dict[str, Any]:
+        """Handle the start of the config flow."""
+
+        self._errors = {}
+
+        if user_input is not None:
+            if await self._test_connection(user_input):
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME], data=user_input
+                )
+        else:
+            user_input = {}
+            user_input[CONF_NAME] = DEFAULT_NAME
+            user_input[CONF_AUTHENTICATION] = HTTP_BASIC_AUTHENTICATION
+            user_input[CONF_LIMIT_REFETCH_TO_URL_CHANGE] = False
+            user_input[CONF_CONTENT_TYPE] = DEFAULT_CONTENT_TYPE
+            user_input[CONF_FRAMERATE] = 2
+            user_input[CONF_VERIFY_SSL] = True
```

Comment:
```suggestion
            user_input = {
               CONF_NAME: DEFAULT_NAME,
               CONF_AUTHENTICATION: HTTP_BASIC_AUTHENTICATION,
               CONF_LIMIT_REFETCH_TO_URL_CHANGE: False,
               CONF_CONTENT_TYPE: DEFAULT_CONTENT_TYPE,
               CONF_FRAMERATE: 2,
               CONF_VERIFY_SSL: True
            }
```

Reply:
There are all defaults so making the whole thing a constant might make sense here as well